WebService/v1/Controllers/DeviceModelScriptsController.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+using Jint.Parser;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Models;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Exceptions;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Filters;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Models;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Models.Helpers;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Models.DeviceModelScriptApiModel;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.WebService.v1.Controllers
+{
+    [ExceptionsFilter]
+    public class DeviceModelScriptsController : Controller
+    {
+        private readonly string ApplicationJavascript = "application/javascript";
+        private readonly ILogger log;
+        private readonly IDeviceModelScripts simulationScriptService;
+
+        public DeviceModelScriptsController(
+            IDeviceModelScripts simulationScriptService,
+            ILogger logger)
+        {
+            this.simulationScriptService = simulationScriptService;
+            this.log = logger;
+        }
+
+        [HttpGet(Version.PATH + "/[controller]")]
+        public async Task<DeviceModelScriptListModel> GetAsync()
+        {
+            return DeviceModelScriptListModel.FromServiceModel(await this.simulationScriptService.GetListAsync());
+        }
+
+        [HttpGet(Version.PATH + "/[controller]/{id}")]
+        public async Task<DeviceModelScriptApiModel> GetAsync(string id)
+        {
+            return DeviceModelScriptApiModel.FromServiceModel(await this.simulationScriptService.GetAsync(id));
+        }
+
+        [HttpPost(Version.PATH + "/[controller]!validate")]
+        public async Task<ActionResult> Validate(IFormFile file)
+        {
+            if (file == null)
+            {
+                this.log.Warn("No data provided");
+                throw new BadRequestException("No data provided.");
+            }
+
+            if (file.ContentType != this.ApplicationJavascript)
+            {
+                this.log.Warn("Wrong content type provided", () => new { file.ContentType });
+                throw new BadRequestException("Wrong content type provided.");
+            }
+
+            try
+            {
+                var parser = new JavaScriptParser();
+                var reader = new StreamReader(file.OpenReadStream());
+                var rawScript = reader.ReadToEnd();
+                parser.Parse(rawScript);
```

Comment:
what do you think about moving these 4 lines into the JavascriptInterpreter class, e.g. adding a ValidateJs method or something similar?

Reply:
Done.